Reactions/ReactorCvode.H

```diff
@@ -51,7 +51,11 @@ class ReactorCvode : public ReactorBase::Register<ReactorCvode>
 public:
   static std::string identifier() { return "ReactorCvode"; }
 
+#if defined(AMREX_USE_GPU) && !defined(PELE_USE_MAGMA)
```

Comment:
MAGMA is not the only matrix-based approach we have on GPU. Having this a runtime parameter based on the choice of linear_solve is not possible ?

Reply:
ooof. That is not what I envisioned when doing this templating way back in the day... I will need to think about that. I guess we need to figure out what the best default is for now?

Reply:
Haven't figured this out yet... One solution I have right now is an if condition around each call where there is an `<ordering>` template argument. So we would have if condtions inside kernel calls. And a whole bunch of 
```
if (ordering == ycorder){
   utils::box_flatten<YCOrder>( icell, i, j, k, ncells, captured_reactor_type, captured_clean_init_massfrac, rY_in, rYsrc_in, T_in, rEner_in, rEner_src_in, yvec_d, udata_g->rYsrc_ext, udata_g->rhoe_init, udata_g->rhoesrc_ext);
}
else{
  utils::box_flatten<CYOrder>( icell, i, j, k, ncells, captured_reactor_type, captured_clean_init_massfrac, rY_in, rYsrc_in, T_in, rEner_in, rEner_src_in, yvec_d, udata_g->rYsrc_ext, udata_g->rhoe_init, udata_g->rhoesrc_ext);
}
```
Not too crazy about this idea but I could be convinced for performance reasons. I think it would lead to larger executables.

Another idea is that we make it a compile flag? Instead of switching on GPU or LM/C we just are explicit about the one we want? We could default it based on other stuff like GPU and LM/C...

Reply:
Adding yet another compile flag sort goes against adding runtime flexibility. I think we can probably make the `if` work but maybe have the if outside of the kernels if possible. Let me have a look.

Reply:
Should we merge this PR and then do that? My guess is that it's going to be some amount of work...

Reply:
In the current state, we break using GMRES if you compile with MAGMA ON for example (assuming we have typical values ON in both cases).

Reply:
Currently, we break the `sparse_direct` solve on CUDA GPUs.

Reply:
That sounds good to me.

Reply:
Ok done in #383